### PR TITLE
Add torchsparse.

### PR DESCRIPTION
**Pull request by benjaminrwilson**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by benjaminrwilson** [comment]
@chrisburr -- have you seen this [build error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=332320&view=logs&j=232b7f5e-8d64-5438-fada-8478f379d226&t=e8516d7e-1153-5090-5807-53ae833fddd8&l=2085) before?

The build appears to be running out of space.

**Comment by benjaminrwilson** [comment]
@conda-forge/staged-recipes , this is failing with [No space left on device](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=333079&view=logs&j=232b7f5e-8d64-5438-fada-8478f379d226&t=e8516d7e-1153-5090-5807-53ae833fddd8&l=2111). Any ideas?

**Comment by ocefpaf** [comment]
@conda-forge-admin, please restart ci

**Comment by chrisburr** [comment]
The easiest solution is probably to disable GPU builds for now and re-enable them in the feedstock (the feedstock CI runs one-job per python version).

File: recipes/torchsparse/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    - {{ compiler('cuda') }}                 # [linux64 and cuda_compiler_version not in (undefined, 'None')]
+  host:
+    - python
+    - pip
+    - sparsehash
+    - pytorch-cpu                            # [osx or cuda_compiler_version in (undefined, 'None')]
+    - pytorch-gpu                            # [cuda_compiler_version not in (undefined, 'None')]
+  run:
+    - python
+    - {{ pin_compatible('pytorch-cpu') }}    # [osx or cuda_compiler_version in (undefined, 'None')]
```
Comment (chrisburr) [inline recipes/torchsparse/meta.yaml:44]: Does pytorch have a stable ABI between minor versions? I suspect you need `max_pin="x.x"` or `max_pin="x.x.x"`.

File: recipes/torchsparse/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    - /lib*/libgcc_s.so.1
+    - /lib*/libc.so.6
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - cython                                 # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
+    - pybind11                               # [build_platform != target_platform]
```
Comment (chrisburr) [inline recipes/torchsparse/meta.yaml:31]: How come these three are only needed when cross compiling?

File: recipes/torchsparse/meta.yaml (hunk 1 above)
Comment (benjaminrwilson) [reply recipes/torchsparse/meta.yaml:44]: Unfortunately, it sounds like the ABI is generally fairly unstable. Here is a brief thread discussing this: https://github.com/conda-forge/pytorch-cpu-feedstock/issues/40#issuecomment-803133327 -- what are your thoughts?

File: recipes/torchsparse/meta.yaml (hunk 2 above)
Comment (benjaminrwilson) [reply recipes/torchsparse/meta.yaml:31]: I was prepping it for https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock -- maybe that's not needed, though?

File: recipes/torchsparse/meta.yaml (hunk 1 above)
Comment (chrisburr) [reply recipes/torchsparse/meta.yaml:44]: I would be tempted to go with `max_pin="x.x.x"` as building pytorch is enough of a pain that builds aren’t that frequent.

File: recipes/torchsparse/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipes/torchsparse/meta.yaml]: It’s probably easiest to add it to the migrator (see the first paragraph)

**Comment by benjaminrwilson** [comment]
@chrisburr , I think this should be good to go now.

**Comment by CCInc** [comment]
Hi @benjaminrwilson , I'm one of the maintainers of torchsparse, awesome work! Would love to get your feedback on how to maintain this going forward, if you could drop us a note in https://github.com/mit-han-lab/torchsparse/issues/93 or shoot me an email that'd be great!

**Comment by benjaminrwilson** [comment]
Hi @CCInc,

Thanks! Would you mind putting up a small ticket for discussion here: https://github.com/conda-forge/torchsparse-feedstock/issues?

**Pull request merged**